mlir/unittests/Analysis/Presburger/MatrixTest.cpp

```diff
@@ -248,3 +248,61 @@ TEST(MatrixTest, computeHermiteNormalForm) {
     checkHermiteNormalForm(mat, hermiteForm);
   }
 }
+
+TEST(MatrixTest, inverse) {
+  FracMatrix mat = makeFracMatrix(
+      2, 2,
+      {{Fraction(2, 1), Fraction(1, 1)}, {Fraction(7, 1), Fraction(0, 1)}});
+  FracMatrix inverse = makeFracMatrix(
+      2, 2,
+      {{Fraction(0, 1), Fraction(1, 7)}, {Fraction(1, 1), Fraction(-2, 7)}});
+
+  FracMatrix invM(2, 2);
+  FracMatrix* inv = &invM;
+  mat.determinant(inv);
+
+  for (unsigned row = 0; row < 2; row++)
+    for (unsigned col = 0; col < 2; col++)
+      EXPECT_EQ(inv->at(row, col), inverse(row, col));
+  
+  mat = makeFracMatrix(
+      2, 2,
+      {{Fraction(0, 1), Fraction(1, 1)}, {Fraction(0, 1), Fraction(2, 1)}});
+  Fraction det = mat.determinant(inv);
+  
+  EXPECT_EQ(det, Fraction(0, 1));
+}
+
+TEST(MatrixTest, intInverse) {
+  IntMatrix mat = makeIntMatrix(2, 2, {{2, 1}, {7, 0}});
+  IntMatrix inverse = makeIntMatrix(2, 2, {{0, -1}, {-7, 2}});
```

Comment:
why is this not a scale of the inverse above?

Reply:
it is, it's scaled by -7, right?

Reply:
ah but shouldn't it be scaled by a positive number, the determinant?